packages/x-codemod/src/v6.0.0/data-grid/rename-linkOperators-logicOperators/index.ts

```diff
@@ -0,0 +1,81 @@
+import { ASTPath, ImportDeclaration } from 'jscodeshift';
+import { JsCodeShiftAPI, JsCodeShiftFileInfo } from '../../../types';
+
+const renamedIdentifiers = {
+  GridLinkOperator: 'GridLogicOperator',
+  linkOperatorInputProps: 'logicOperatorInputProps',
+  linkOperator: 'logicOperator',
+  linkOperators: 'logicOperators',
+  setFilterLinkOperator: 'setFilterLogicOperator',
+};
+
+const renamedLiterals = {
+  filterPanelLinkOperator: 'filterPanelLogicOperator',
+};
+
+const renamedClasses = {
+  'MuiDataGrid-filterFormLinkOperatorInput': 'MuiDataGrid-filterFormLogicOperatorInput',
+  'MuiDataGrid-withBorder': 'MuiDataGrid-withBorderColor',
+};
+
+const PACKAGE_REGEXP = /@mui\/x-data-grid(-pro|-premium)?/;
+
+const matchImport = (path: ASTPath<ImportDeclaration>) =>
+  ((path.node.source.value as any).toString() || '').match(PACKAGE_REGEXP);
+
+// Press ctrl+space for code completion
+export default function transform(file: JsCodeShiftFileInfo, api: JsCodeShiftAPI, options: any) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  const printOptions = options.printOptions || {
+    quote: 'single',
+    trailingComma: true,
+  };
+
+  const matchingImports = root.find(j.ImportDeclaration).filter((path) => !!matchImport(path));
+
+  if (matchingImports.length > 0) {
+    // Rename the identifiers
+    // - import { linkOperatorInputProps } from '@mui/x-data-grid'
+    // + import { logicOperatorInputProps } from '@mui/x-data-grid'
+    root
+      .find(j.Identifier)
+      .filter((path) => !!renamedIdentifiers[path.node.name])
+      .replaceWith((path) => j.importSpecifier(j.identifier(renamedIdentifiers[path.node.name])));
+
+    // Rename the literals
+    // - apiRef.current.getLocaleText('filterPanelLinkOperator')
+    // + apiRef.current.getLocaleText('filterPanelLogicOperator')
+    root
+      .find(j.Literal)
+      .filter((path) => !!renamedLiterals[path.node.value as any])
+      .replaceWith((path) => j.literal(renamedLiterals[path.node.value as any]));
+
+    // Rename the classes
+    // - 'MuiDataGrid-filterFormLinkOperatorInput'
+    // + 'MuiDataGrid-filterFormLogicOperatorInput'
```

Comment:
Have you considered/tried using `renameClassKey` utility for this transform? 🤔

Reply:
Yes, actually it doesn't work great in out use-case as it's targeting single component and `classes` prop. For our usecase we want to handle `sx` prop and legacy styling like this too:
https://github.com/mui/mui-x/blob/e0d7f4101f1b9df484687658f4db9bdbf4e40e8d/packages/x-codemod/src/v6.0.0/data-grid/rename-linkOperators-logicOperators/actual.spec.js#L5-L10

Will try to add this new (more generic) one to the util or replace the other with this.

Reply:
Yeah, by no means, feel free to update the existing util with the required logic.
I've only left the existing util in the codebase after cloning MUI Core codemod, because I thought that we'd have use for it, but so far it hasn't been used once. 🙈